natlib/svcd.c

```diff
@@ -185,3 +191,119 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+//////////////////////////////////////////////////////////////////////////////
+// SVCD.write((string targetip, number svcid, number attrid, lightfunction payload, 
+//                        number timeout_ms, lightfunction on_done)) 
+// Authors: Aparna Dhinakaran, David Ho, Romi Phadte, Cesar Torres
+/////////////////////////////////////////////////////////////
+
+
+void resolve_table(lua_State *L, char* key){
+    // push the global table, resolve it and then pop it off the stack
+    lua_getglobal(L, "SVCD");
+    lua_pushstring(L, key);
+    lua_gettable(L, -2);
+    lua_insert(L, -2);
+    lua_settop(L, -2);
+}
+
+static int svcd_write( lua_State *L )
+{   
+
+    int numargs = lua_gettop(L); // 6
+		if (numargs == 6) {
+			/* GET ALL THE ARGS */
+			size_t g;
+			const char* targetip = (char*) lua_tolstring(L, 1, &g);
+			int svcid = (int)lua_tonumber(L, 2);
+			int attrid = (int)lua_tonumber(L, 3);
+ 
+			size_t l;
+			const char* payload =  lua_tolstring(L, 4, &l);
+
+			int timeout_ms = (int)lua_tonumber(L, 5);
+			int* on_done = (int*) lua_tonumber(L, 6);
+    
+			/* PARAMS */
+    
+			/* SET IVKID */
+			resolve_table(L, "ikvid"); //7
+			int ivkid = (int) lua_tonumber(L, -1);
+   
+  
+			ivkid =  ivkid + 1;
+			if( ivkid > 65535 ) { 
+					ivkid = 0;
+			}
+
+			lua_settop(L, numargs);
+
+			// setting ivkid
+			lua_getglobal(L, "SVCD");
+			lua_pushnumber(L, ivkid);
+			lua_setfield(L, -2, "ikvid");
+			lua_settop(L, numargs);
+			/* END SET IVKID*/
+ 
+			/* SET HANDLER */
+			resolve_table(L, "handlers"); // 7
+
+			lua_pushnumber(L, ivkid);
+			lua_pushlightfunction(L, on_done);
+
+			lua_settable(L, -3);
+			/* END SET HANDLER */
+    
+			/* LUA WRITE INVOKE */
+			resolve_table(L, "write_invoke_later"); // 9
+			//function write invoke later 
+					lua_pushnumber(L, ivkid);  // 10
+					lua_pushnumber(L, timeout_ms);
+					lua_call(L, 2, 0);
+			/* END LUA WRITE INVOKE */
+
+
+			/* SEND TO */
+			lua_pushlightfunction(L, libstorm_net_sendto); 
+			// resolve_table(L, "print_sendto"); 
```

Comment:
You can remove this line; It was a special debugging statement. Was resolving a lua function ```SVCD.print_sendto```...